astropy/table/table.py

```diff
@@ -3857,6 +3857,13 @@ class QTable(Table):
     except that columns with an associated ``unit`` attribute are converted to
     `~astropy.units.Quantity` objects.
 
+    .. note:: When a column of ``int`` ``dtype`` is converted to ``Quantity``, its
+              ``dtype`` is converted to ``float``.
+              For example, for a quality flag column of ``int``, if it is
+              assigned with the unit ``unitless``, it will still be converted to
```

Comment:
suggesting linking to the corresponding unit.

```suggestion
              assigned with the unit `~astropy.units.dimensionless_unscaled`, it will still be converted to
```

Reply:
Resolved 👍